### PR TITLE
feat: adding set_once feature for Mixpanel

### PR DESCRIPTION
**Pull request by shrouti1507**
## PR Description

Adding set_once property for mixpanel
resolves INT-948

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js [hunk 1]
```diff
@@ -131,7 +133,9 @@ class Mixpanel {
     // determine which traits to union to existing properties and which to set as new properties
     const traitsToUnion = {};
     const traitsToSet = {};
+    const traitsToSetOnce = {};
     Object.keys(traits).forEach(trait => {
+      if (!setOnceProperties.includes(trait)) {
```
Comment (utsabc) [inline packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:138]: Please format this file

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js [hunk 2]
```diff
@@ -146,8 +150,16 @@ class Mixpanel {
...
       }
+    } else {
+      // as this set is not supposed to be set any more in the future
+      traitsToSetOnce[trait] = traits[trait];
+    }
     });
-
+    
+    if(traitsToSetOnce && Object.keys(traitsToSetOnce).length > 0) {
+      window.mixpanel.people.set_once(traitsToSetOnce);
```
Comment (utsabc) [inline packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:160]: can you link Mixpanel documentation?

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js (hunk 1 above)
Comment (shrouti1507) [reply packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js]: done

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js (hunk 2 above)
Comment (shrouti1507) [reply packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js]: done

**Review by mihir-4116 — DISMISSED** [review]

File: packages/analytics-js-integrations/__tests__/integrations/Mixpanel/util.test.js [hunk 3]
```diff
@@ -213,3 +215,226 @@ describe('mapTraits', () => {
...
+    expect(result.setOnce).toEqual({
+      street: '123 Main St',
+      city: 'New York',
+    });
+    expect(result.email).toBe('test@example.com');
+    expect(result.username).toBeUndefined();
+  });
+});
+
+// Generated by CodiumAI
```
Comment (Gauravudia) [inline packages/analytics-js-integrations/__tests__/integrations/Mixpanel/util.test.js:407]: remove this.

Comment (shrouti1507) [reply packages/analytics-js-integrations/__tests__/integrations/Mixpanel/util.test.js]: done

**Review by mihir-4116 — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/Mixpanel/util.js [hunk 4]
```diff
@@ -32,7 +33,71 @@ const traitAliases = {
...
+ * unset(obj, 'person.address.city');
+ *  Output: { person: { name: 'John', age: 30, address: { state: 'NY' } } }
+ */
+function unset(obj, propertyPath) {
+  const keys = propertyPath.split('.');
+  const lastKey = keys.pop();
+
+  let current = obj;
+  for (const key of keys) {
+    if (current[key] === undefined || current[key] === null) {
```
Comment (Gauravudia) [inline packages/analytics-js-integrations/src/integrations/Mixpanel/util.js:64]: can't we use `if (!current[key]) {}`?

**Review by Gauravudia — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/Mixpanel/util.js [hunk 5]
```diff
@@ -32,7 +33,71 @@ const traitAliases = {
...
+  const setOnceEligible = {};
+
+  // Step 1: find the k-v pairs of setOnceProperties in traits and contextTraits
+
+  setOnceProperties.forEach((propertyPath) => {
+    const pathSegments = propertyPath.split('.');
+    const propName = pathSegments[pathSegments.length - 1];
+
+    if (Object.keys(traitsCopy).length > 0 && get(traitsCopy, propertyPath)) {
+      setOnceEligible[propName] = get(traitsCopy, propertyPath);
```
Comment (ItsSudip) [inline packages/analytics-js-integrations/src/integrations/Mixpanel/util.js:87]: As the propertyPath will contain the exact path of the trait can we run it on top of the payload apart from running on traitsCopy?

Comment (shrouti1507) [reply packages/analytics-js-integrations/src/integrations/Mixpanel/util.js:87]: no we need not. Reason being :
-  this is identify event so we will expect traits only not properties. 
-  If we follow our already [existing code](https://github.com/rudderlabs/rudder-sdk-js/blob/89b216283752b53d272e3015dc7b644f94799211/packages/analytics-js-integrations/src/integrations/Mixpanel/util.js#L101) we are just fetching traits from context only. 
- The property path concept has come for any sub-properties, like for `address.city` and `address.country` etc where we need the exact path of the particular property in payload to fetch its values. Will raise doc card accordingly

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**